### PR TITLE
Allow facilities to be filtered by name

### PR DESCRIPTION
**Pull request by jcmorrow**
The actual search part of this change was pretty easy - it's a text box that updates the list of visible facilities onChange. The hard part was that we were storing which facility was selected by *index* for some reason, rather than id. This means that if you had selected the 2nd of 3 facilities, and then you filtered out the first one via search, it would look like the selected facility had *changed*. I've changed this to keep track via the id of the selected facility, but to be perfectly honest the code around keeping track of that state looks like a huge mess, and I don't really want to bother trying to understand it thoroughly. I suspect that the correct long-term change for simplicity here would be to pull out the index tracking completely, but I can't tell all of the places that the index tracking is being used, so for right now I haven't done that.

### CODE REVIEW

File: src/components/task_list.tsx [hunk 1]
```diff
@@ -100,17 +103,38 @@ class TaskList extends React.Component<Props, State> {
...
   }
 
+  _filterTaskGroups = (
+    taskGroups: TaskGroup[],
+    searchTerm?: string
+  ): TaskGroup[] => {
+    if (searchTerm === undefined || searchTerm === "") {
+      return taskGroups;
+    } else {
+      return taskGroups.filter(t => t.site.name.match(searchTerm));
```
Comment (liangmicha) [inline src/components/task_list.tsx:113]: Does `match` does regex matching, or substring matching, or direct string match?

Comment (liangmicha) [reply src/components/task_list.tsx:113]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match#a_non-regexp_object_as_the_parameter

It seems like `match` returns a list of strings, and I'm guessing filter knows how to interpret that as a `true` boolean?

**Review by liangmicha — APPROVED** [review]
Looks good

File: src/components/task_list.tsx (hunk 1 above)
Comment (jcmorrow) [reply src/components/task_list.tsx:113]: It does regex matching, and if you pass a string to it (as we are here) your string is automatically converted to a regex (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match)

**Pull request merged**